core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFormatterUtil.java

```diff
@@ -201,6 +295,102 @@ static ExprValue getFormattedTime(ExprValue timeExpr, ExprValue formatExpr) {
     return getFormattedString(formatExpr, TIME_HANDLERS, time);
   }
 
+  private static ExprCoreType getReturnType(ExprValue formatStringExpr) {
+    String formatString = formatStringExpr.stringValue();
+
+    boolean hasDate = false;
+    boolean hasTime = false;
+
+    for (String dateSpecifier: DATE_SPECIFIERS) {
+      if (formatString.contains(dateSpecifier)){
+        hasDate = true;
+      }
+    }
+    for (String timeSpecifier: TIME_SPECIFIERS) {
+      if (formatString.contains(timeSpecifier)){
+        hasTime = true;
+      }
+    }
+
+    if (hasDate && hasTime){
+      return ExprCoreType.DATETIME;
+    } else if (hasTime) {
+      return ExprCoreType.TIME;
+    }
+
+    return ExprCoreType.DATE;
+  }
+
+  static private boolean checkValidTemporalAccessor(ExprCoreType type, TemporalAccessor ta) {
```

Comment:
Proposal
1. Let this function to get only `TemporalAccessor` as an argument. It should return `true` if `ta` has data to build complete `DATETIME`, `TIME`, or `DATE` objects (no change in logic). Outer function should return `null` if this one returns `false`.
2. Replace unsupported fields by default values[^1] and build full `DATETIME` object.


[^1]: 1 for month and day and 0 for all rest.

Reply:
I have attempted to do fix 1. Much of the logic is still similar, but some refactoring had to occur. Also, as a result of ensuring the function returns a `Datetime`, default values had to be used to fill in. extra fields if enough information was only given to make either a `Date` or `Time`.